Modules/Quotas/src/Quotas.Application/Tiers/Commands/CreateQuotaForIdentity/Handler.cs

```diff
@@ -32,6 +33,11 @@ public async Task<IndividualQuotaDTO> Handle(CreateQuotaForIdentityCommand reque
 
         var metric = await _metricsRepository.Find(parseMetricKeyResult.Value, cancellationToken);
 
+        if (identity.IndividualQuotas.Any(q => q.MetricKey == metric.Key && q.Period == request.Period))
+        {
+            throw new DomainException(DomainErrors.DuplicateQuota());
+        }
```

Comment:
Once again: please put as much logic as possible into the Domain classes.

Reply:
Still not what I expected. 
The goal is that it is impossible for the user of the Identity class to do something that leads to an invalid Identity (may it be on purpose or not). Currently you could remove the call of the `IndividualQuotaAlreadyExists` method.

So to solve this, simply move the check into the `CreateIndividualQuota` method.

Reply:
That makes sense. I came across a weird test however, which I had to change. Please ensure that change makes sense.

https://github.com/nmshd/backbone/blob/d7d2ad833f06a1669bb4c37d288bd9ed1b452296/Modules/Quotas/test/Quotas.Application.Tests/Tests/Quotas/DeleteTierQuotaDefinition/HandlerTests.cs#L57

Reply:
That test needs to be updated, of course. :)